ext/date/tests/bug81500.phpt

```diff
@@ -0,0 +1,16 @@
+--TEST--
+Bug #81500 (Interval serialization regression since 7.3.14 / 7.4.2)
+--FILE--
+<?php
+$interval = new DateInterval('PT1S');
+$interval->f = -0.000001;
+var_dump($interval->s, $interval->f);
+
+$interval = unserialize(serialize($interval));
+var_dump($interval->s, $interval->f);
+?>
+--EXPECT--
+int(1)
+int(-1)
```

Comment:
Why is it `int(-1)` while we asked for `float(-0.000001)`? 🤔

Reply:
Negative values are not supposed to be supported; `-1` marks invalid µs.

Reply:
Sorry but `-1` is not a proper way to mark something as invalid, just throw an exception if it's forbidden. Here it will just become muted and unnoticed then doing calculation with it assuming it's a relevant value.

Reply:
PHP 7 behavior just completely feel like the correct behavior, you pass a value to `->f` serialize/unserialize and find there what you passed, exactly like any other units still do in PHP 8: https://3v4l.org/r0i54

Why would `->f` would be a particular case and proceed this insane any-out-of-range-value to `-1` transformation?